config/docker/build-and-push.sh

```diff
@@ -100,6 +102,18 @@ then
   fi
 fi
 
+if [ "x${buildroot}" == "xtrue" ]
+then
+  tag=${tag_px}buildroot
+  echo_build $tag
+  docker build ${quiet} ${cache_args} debos -t $tag
```

Comment:
Erm, that's blatant copy-paste mistake.  Please try the script before adding a change to the PR.

Reply:
You're right I made copy-pasted the part of the script for debos and changed the tag name, but haven't actually changed the Dockerfile to use.
I did have run the script before adding changes to the PR. I built the image and it was tagged properly, I haven't spotted though it's not the right image. My fault.

Reply:
I've added the `buildroot` images to be built automatically on staging, so if you update this PR it should get built later today and then you can pull the image `kernelci/staging-buildroot` and check it works.

Reply:
Refactored `build-and-push.sh` script. I got rid of the redundant code blocks and extracted them to functions. It should make it easier to add new images in future and improve readability of the script.

I tested it locally with the following calls, which are the same as in `kernelci-deploy/kernelci.org` script.

```
./build-and-push.sh -drn
./build-and-push.sh -bikQn
```
I couldn't spot any misbehaviour. 

**Pushing to docker hub has not been tested.**

build logs attached
[build-cmd_docker.log](https://github.com/kernelci/kernelci-core/files/6800620/build-cmd_docker.log)
[build-cmd_rootfs_docker.log](https://github.com/kernelci/kernelci-core/files/6800622/build-cmd_rootfs_docker.log)